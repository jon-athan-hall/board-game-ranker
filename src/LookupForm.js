import React, { Component } from "react";

class LookupForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: 'username'
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({
      username: e.target.value
    });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit} className="flex flex-col p-8 space-y-4">
        <label htmlFor="username">BoardGameGeek username</label>
        <input
          type="text"
          id="username"
          value={this.state.username}
          onChange={this.handleChange}
          className="p-4 border border-gray-500 rounded-md"
        />
        <button type="submit" className="p-4 bg-yellow-300 rounded-md">
          {this.props.buttonText}
        </button>
      </form>
    );
  }
}

export default LookupForm;
