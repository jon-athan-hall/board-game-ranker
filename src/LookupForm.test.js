import React from 'react';
import renderer from 'react-test-renderer';
import LookupForm from './LookupForm';

it('renders correctly', () => {
  const tree = renderer.create(<LookupForm />).toJSON();
  expect(tree).toMatchSnapshot();
});

/**
 * expect();
 * .toBe(n);
 * .toEqual(n);
 * .not.toBe(n);
 * .toBeDefined();
 * .toBeFalsy();
 * .toBeGreaterThanOrEqual(n);
 * .toMatch('string');
 * .toContain(value);
 * .toThrow(error);
 * done(); to wait for callback results.
 * use return
 */