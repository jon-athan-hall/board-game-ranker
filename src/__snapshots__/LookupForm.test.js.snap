// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="flex flex-col p-8 space-y-4"
>
  <label
    for="username"
  >
    BoardGameGeek username
  </label>
  <input
    className="p-4 border border-gray-500 rounded-md"
    id="username"
    type="text"
  >
    username
  </input>
  <button
    className="p-4 bg-yellow-300 rounded-md"
    type="submit"
  />
</form>
`;
